File: Sort_algo/bubble_sort.py
arr = []
def bubbleSort(arr):
    n = len(arr)
    for i in range(n):
        swapper = False
        for j in range(0,n-(i+1)):
            if arr[j] > arr[j+1]:
                arr[j],arr[j+1] = arr[j+1],arr[j]
                swapper = True
        if swapper == False:
            break
def input_arr():
    n = int(input("Enter size of array:\n"))
    for i in range(0,n):
        ele = int(input("Enter Elements: \n"))
        arr.append(ele)
input_arr()
print("Before\n")
print(arr)
bubbleSort(arr)
print("after\n")
print(arr)

File: Sort_algo/bubbleSort.cpp
#include<bits/stdc++.h>
using namespace std;
void swapping(int *xp,int *yp)
{
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}
void bubbleSort(int arr[],int n)
{
    bool swapper;
    for(int i=0;i<n-1;i++)
    {
        swapper = false;
        for(int j=0;j<n-(i+1);j++)
        {
            if(arr[j] > arr[j+1])
            {
                swapping(&arr[j],&arr[j+1]);
                // swap( arr[j],arr[j+1] );
                swapper = true;
            }
        }
        if(swapper == false)
            break;
    }
}
void printArray(int arr[],int n)
{
    for(int i=0;i<n;i++)
    {
        cout << arr[i] << " " ;
    }
    cout << "\n";
}
int main()
{
    int n;
    cout << "Enter the size of array: \n";
    cin >> n;
    int arr[n];
    cout << "Enter the Elements of array: \n";
    for(int i=0;i<n;i++)
    {
        cin >> arr[i];
    }
    cout << "Array before getting sorted : \n";
    printArray(arr,n);
    bubbleSort(arr,n);
    cout << "Array after getting sorted : \n";
    printArray(arr,n);
}